mitodl/micromasters
# feat: add django-robots for dynamic robots.txt

#### Pre-Flight checklist


#### What are the relevant tickets?
#5267 

#### What's this PR do?
- Removes the static `robots.txt`
- Adds [django-robots](https://pypi.org/project/django-robots/) 
- Adds an initial migration to pre-populate the robots.txt file with values from the existing one
- Adds settings for `BASE_URL, ROBOTS_CACHE` etc
- Enables Django's sites

#### How should this be manually tested?
- First, notice the existing `robots.txt` file in the project
- Add a value in .env for `MICROMASTERS_BASE_URL=http://mm.odl.local:8079`
- Move to this branch then `docker-compose build` followed by `docker-compose up`, notice the `robots.txt` file doesn't exist anymore in the static directory
- Upon moving this PR the build should be successful
- Check that a default website is added in `sites` in Django Admin with a value that matches `MICROMASTERS_BASE_URL` in your .env
- Check the `Rules` model under `Robots` in Django Admin and notice a rule exists with `*` (for all types of robots)
- Check that `Url` and check that it has existing disallowed URLs in that
- Check `localhost/robots.txt`  you should see these added rules in that
- Add a `Url` manually, add that to the Rule and notice that it's reflected in `localhost/robots.txt`

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
The PR and its contents are inspired by https://github.com/mitodl/mitxpro/pull/681

#### Screenshots (if appropriate)
<img width="1625" alt="image" src="https://user-images.githubusercontent.com/34372316/211323749-481b6dda-6fd1-4318-b7ad-dff100516752.png">
<img width="1651" alt="image" src="https://user-images.githubusercontent.com/34372316/211323820-f04afa58-c7be-4103-a3cc-84de1c763b9c.png">
<img width="798" alt="image" src="https://user-images.githubusercontent.com/34372316/211323898-1f3b869d-6cab-4b8a-9dac-b8669cf2b324.png">
<img width="489" alt="image" src="https://user-images.githubusercontent.com/34372316/211323948-c944c0f1-d599-45fa-8eca-69c8bdb9f9da.png">


#### What GIF best describes this PR or how it makes you feel?
(Optional)
